### PR TITLE
bpo-46566: Add new py.exe launcher implementation

### PR DESCRIPTION
**Pull request by zooba**
https://bugs.python.org/issue46566

### CODE REVIEW

File: PC/launcher2.c [hunk 1]
```diff
@@ -0,0 +1,2165 @@
...
+}
+
+
+USHORT
+_getNativeMachine()
+{
+    static USHORT _nativeMachine = IMAGE_FILE_MACHINE_UNKNOWN;
+    if (_nativeMachine == IMAGE_FILE_MACHINE_UNKNOWN) {
+        USHORT processMachine;
+        if (!IsWow64Process2(NULL, &processMachine, &_nativeMachine)) {
```
Comment (conioh) [inline PC/launcher2.c]: Will that load on Windows 8?

Comment (zooba) [reply PC/launcher2.c]: No, it won't. I'll have to do it dynamically for Win 8.1 users (Win 8 itself is EOL).

Comment (conioh) [reply PC/launcher2.c]: According to the documentation, though I haven't checked myself, it's also absent in Windows 10 version 1507 which is still under extended support in the LTSB branch.

In the fix for this I'd have change the `typedef` to something like `using PIsWow64Process2 = decltype(&IsWow64Process2);`
It's not critical and may be called a matter of style, but I see no benefit from providing the types yourself.

**Comment by sweeneyde** [comment]
Could this be related to the Azure Pipelines CI trouble?

Specifically `Azure Pipelines CI`/`Windows CI Tests win64`/`appx Tests`/`test_launcher` is failing on the last couple of commits to main.

**Comment by nsait-linaro** [comment]
@zooba `test_launcher` seems to be failing for win/arm64 after this patch

https://buildbot.python.org/all/#/builders/730/builds/4306

**Comment by zooba** [comment]
> Could this be related to the Azure Pipelines CI trouble?

Yes, the test doesn't account for finding an old version of the launcher already installed. I'll fix.

> `test_launcher` seems to be failing for win/arm64 after this patch

I can't tell what's going wrong there, and it doesn't repro on my own ARM64 machine. I'll file a new bug for this one

**Pull request merged**